src/main/java/seedu/address/commons/util/StringUtil.java

```diff
@@ -72,15 +72,15 @@ public static boolean isNonZeroUnsignedInteger(String s) {
     /**
      * Returns true if {@code s} represents a valid numeric double precision floating point value
      * Will return false for any other non-null string input
-     * e.g. empty string, "-1", "0", "+1", "3 0" (contains whitespace), "1 a" (contains letters)
+     * e.g. empty string, "3 0" (contains whitespace), "1 a" (contains letters)
      * @throws NullPointerException if {@code s} is null.
      */
     public static boolean isValidNumber(String s) {
         requireNonNull(s);
 
         try {
             double value = Double.parseDouble(s);
-            return !(value < 0) && !s.matches(".*[-+\\p{Alpha}].*");
```

Comment:
Negative allowed for adding?

Reply:
It is possible that the dollars made from a coin account is negative. i.e. Buying coins before selling them. Users may want to know on which accounts they have yet to turn a profit and how far behind they are before doing so. For accounting purposes.